clients/src/test/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumerTest.java

```diff
@@ -463,6 +466,44 @@ public void onPartitionsAssigned(final Collection<TopicPartition> partitions) {
         assertTrue(callbackExecuted.get());
     }
 
+    @Test
+    public void testCheckForNewTopicOnlyWhenMetadataChange() {
```

Comment:
maybe clearer to rename this so it shows that it's about subscription regex eval and poll? what about `testSubscriptionRegexEvalOnPollOnlyIfMetadataChanges`?